chcemvediet/apps/inforequests/admin.py

```diff
@@ -385,20 +389,69 @@ def get_inforequest(self, obj):
         return obj.branch.inforequest
 
     def delete_constraints(self, obj):
-        dependency = []
+        constraints = []
         if obj.type in [Action.TYPES.REQUEST, Action.TYPES.ADVANCED_REQUEST]:
-            dependency.append(format_html(
+            constraints.append(format_html(
                 u'{} is type {}.'.format(admin_obj_format(obj), obj.get_type_display())))
         if len(obj.branch.actions) == 1:
-            dependency.append(format_html(
+            constraints.append(format_html(
                 u'{} is the only action in the branch.'.format(admin_obj_format(obj))))
-        return dependency
+        return constraints
 
     def render_delete_form(self, request, context):
         context[u'delete_constraints'] = self.delete_constraints(context[u'object'])
         context[u'ADMIN_EXTEND_SNOOZE_BY_DAYS'] = ADMIN_EXTEND_SNOOZE_BY_DAYS
         return super(ActionAdmin, self).render_delete_form(request, context)
 
+    @transaction.atomic
+    def delete_selected(self, request, queryset):
+        constraints = []
+        warnings = []
+        outbound = InforequestEmail.objects.none()
+        inbound = InforequestEmail.objects.none()
+        for obj in queryset:
+            obj_constraints = self.delete_constraints(obj)
+            constraints += obj_constraints
+            inforequestemails = self.nested_inforequestemail_queryset(obj)
+            outbound |= inforequestemails[0]
+            inbound |= inforequestemails[1]
+            if not obj_constraints:
+                if not obj.is_last_action and obj.next_action not in queryset:
+                    warnings.append(format_html(squeeze(u"""
+                            {} is not the last action in the branch. Deleting it may cause logical
+                            errors in the inforequest history.
+                            """).format(admin_obj_format(obj))))
+
+        if request.POST.get(u'post'):
+            if constraints:
+                raise PermissionDenied
+
+        template_response = delete_selected(self, request, queryset)
+
+        if request.POST.get(u'post'):
+            n = outbound.count()
+            if n:
```

Comment:
Fungovalo by to aj bez `n`, ale asi nechceme vypis na obrazovke typu "Uspesne vymazanych 0 inforequestemailov"

Reply:
A treba vobec userovi explicitne zobrazovat message, ze sa zmazali instancie inforequestemailov? Pri klasickom delete nic take nezobrazujeme, ci? Nestaci vypisat, kolko sa zmazalo instacii `Action`? A brat ako samozrejme, ze s nimi sa zmazali aj vsetky related instancie, ktore boli pri confirm vypisane?

Ako funguje built-in bulk delete funkcia? Ak mame model A na ktory je FK z modelu B, teda pri mazani instancii A sa mazu aj instancie B, tak pri built-in bulk delete sa zobrazi message iba o mazani instacii A, alebo sa vypisu dva messages, jedna o mazani A a druha o mazani B?

Reply:
Nie netreba, na obrazovku sa vypise iba pocet oznacenych typu A. V nasom pripade ak mame oznacene napr. 2 akcie, na ktore su naviazane dalsie 4 akcie (spolu 6), na admin obrazovke sa vypise, ze boli uspesne vymazane 2 akcie.